tests/integration_tests/test_collection_query.py

```
@@ -0,0 +1,241 @@
+                embedding = embedding[:dimension]
+            
+            # Convert vector to string format: [1.0,2.0,3.0]
+            vector_str = "[" + ",".join(map(str, embedding)) + "]"
+            # Convert metadata to JSON string
+            metadata_str = json.dumps(data["metadata"], ensure_ascii=False).replace("'", "\\'")
+            # Escape single quotes in document
+            document_str = data["document"].replace("'", "\\'")
+            
+            # Use CAST to convert string to binary for varbinary(512) field
+            sql = f"""INSERT INTO `{table_name}` (_id, document, embedding, metadata) 
+                     VALUES (CAST('{id_str_escaped}' AS BINARY), '{document_str}', '{vector_str}', '{metadata_str}')"""
```

Ruff (0.14.8)

Possible SQL injection vector through string-based query construction

(S608)